packages/registry/docs/conceptual-guides/add-on-loader.md

```diff
@@ -0,0 +1,91 @@
+---
+myst:
+  html_meta:
+    "description": "An explanation of the add-ons configuration loader in @plone/registry"
+    "property=og:description": "An explanation of the add-ons configuration loader in @plone/registry"
+    "property=og:title": "Add-ons configuration loader"
+    "keywords": "@plone/registry, registry, add-ons, loader"
+---
+
+# Add-ons configuration loader
+
+The `@plone/registry` compatible add-ons are able to load configuration into the configuration registry.
+This is achieved by using the `main` key entry point specified in the add-on {file}`package.json`.
+This should be a JavaScript or TypeScript file, eg. {file}`index.ts` or {file}`index.js` placed somewhere (generally in the root) of your add-on.
+
+```json
+{
+  "main": "index.ts",
+}
+```
+
+This file should contain a default export with a function with this signature:
+
+```ts
+import type { ConfigType } from '@plone/registry';
+
+export default function loadConfig(config: ConfigType) {
+  // You can mutate the configuration object in here
+  return config;
+}
+```
+
+`@plone/registry` has a mean to generate an add-ons loader file that contains the code needed to load the add-ons configuration of all the registered add-ons, keeping the order in which they were defined.
+
+This loader is a JavaScript file and is placed in the root of your application.
+By default, it's called {file}`.registry.loader.js`.
```

Comment:
I would avoid starting it with a dot. It makes it a hidden file, so people will miss it.

Reply:
A bit of history, this used to be a hidden file in the /tmp folder so people can't modify it. It's like this in current Volto. It's generated and managed by `@plone/registry`. Since Vite forces you to have this addressable in the app repo boundaries, I opted by use dots to mark it as "special". I did the same for the new style loader. Does this make sense?